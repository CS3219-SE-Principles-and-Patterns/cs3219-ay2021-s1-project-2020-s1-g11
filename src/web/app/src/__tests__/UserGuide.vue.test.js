import {createLocalVue, shallowMount} from '@vue/test-utils'
import ElementUI from 'element-ui';
import VueRouter from 'vue-router';
import Vuex from 'vuex';
import UserGuide from "./../views/UserGuide.vue";

const localVue = createLocalVue();

localVue.use(Vuex);
localVue.use(VueRouter);
localVue.use(ElementUI);

describe("UserGuide.vue", () => {
  let store;

  beforeEach(() => {
    store = new Vuex.Store({
      state: {
        isPageLoading: false,
        userInfo: {
          isApiError: false,
        },
        dbMetaData: {
          entitiesStatus: {
            isLoading: false,
          }
        }
      }
    })
  });

  it('renders correctly', () => {
    const wrapper = shallowMount(UserGuide, {store, localVue})
    expect(wrapper.element).toMatchSnapshot()
  });
});