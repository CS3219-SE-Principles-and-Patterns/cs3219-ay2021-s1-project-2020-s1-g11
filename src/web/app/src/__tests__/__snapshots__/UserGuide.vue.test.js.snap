// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserGuide.vue renders correctly 1`] = `
<div>
  <div>
    <el-container-stub>
      <el-main-stub>
        <el-tabs-stub
          tabposition="top"
        >
          <el-tab-pane-stub
            label="Overview"
          >
            <overviewguide-stub />
          </el-tab-pane-stub>
           
          <el-tab-pane-stub
            label="Features"
          >
            <featureguide-stub />
          </el-tab-pane-stub>
           
          <el-tab-pane-stub
            label="Instructions"
          >
            <instructionsguide-stub />
          </el-tab-pane-stub>
           
          <el-tab-pane-stub
            label="FAQ"
          >
            <faqguide-stub />
          </el-tab-pane-stub>
        </el-tabs-stub>
      </el-main-stub>
    </el-container-stub>
  </div>
</div>
`;
